src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -89,8 +89,10 @@ abstract class RegistryFunctions {
   def tnum(name: String): TVariable =
     tv(name, _.isInstanceOf[TNumeric])
 
-  def registerCode(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitMethodBuilder, Array[Code[_]]) => Code[_]) {
+  def registerCode(mname: String, aTypes: Array[Type], rType: Type, isDet: Boolean)(impl: (EmitMethodBuilder, Array[Code[_]]) => Code[_]) {
```

Comment:
`isDeterministic` more consistent with everywhere else

Reply:
I think this causes some weird shadowing to happen when I do 
```
val isDeterministic = isDeterministic
```
inside the IR definition, which is why I named it differently (the second `isDeterministic` tries to pick up the IRFunction.isDeterministic that we're trying to define, and everything ends up false.) I wasn't sure how to fix this problem without renaming the argument.